py_zipkin/zipkin.py

```diff
@@ -157,21 +157,19 @@ def __enter__(self):
         never attached in the unsampled case, so the spans are never logged.
         """
         self.do_pop_attrs = False
-        # If this span is the first span to be recorded for a service, then
-        # logging will need to be set up.
-        self.is_root = False
+        # If zipkin_attrs are passed in or this span is doing its own sampling,
+        # it will need to actually log spans at __exit__.
+        self.perform_logging = self.zipkin_attrs or self.sample_rate is not None
 
-        if self.zipkin_attrs:
-            self.is_root = True
         if self.sample_rate is not None:
-            self.is_root = True
-
+            # This clause allows for sampling this service independently
```

Comment:
if so i think it'll be worth mentioning this in the docstring for the sample_rate param